src/components/Card.js

```diff
@@ -26,13 +39,42 @@ export default class Card {
   }
 
   _handleLikeIcon() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
+    const cardId = this._cardData._id;
+    const likeButton = this._cardElement.querySelector(".card__like-button");
+    const isLiked = likeButton.classList.contains("card__like-button_active");
+
+    if (cardId.isLiked) {
+      api
+        .dislikeCard(cardId)
```

Comment:
Needs correction: the card should not know about the api, all the logic should be in index.js

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```

Reply:
I am struggling a lot with this, I have done the logic for the like rendering but when I try to do the logic in index.js everything breaks. I can't simply do card.isLiked() because card is only defined in my createCard function and I don't know how to broaden its scope or implement the logic inside that function without breaking my code

Reply:
The like handler can accept the card instance as the parameter